pkgs/development/python-modules/tensorflow/default.nix

```diff
@@ -105,11 +109,32 @@ let
     };
 
     buildAttrs = {
+      # Here we use `declare` to solidify the environment when cc/cxx/ld is called
+      # It's gross but it works!
+
+      # We also set CC to `CXX || CC` since Bazel invokes CC for C++ files occasionally
+      # dont export shellopts. it errors idk why :/
       preBuild = ''
         patchShebangs .
         find -type f -name CROSSTOOL\* -exec sed -i \
           -e 's,/usr/bin/ar,${binutils.bintools}/bin/ar,g' \
           {} \;
+
+        export CC="$(pwd)/cc.sh"
+        export CXX="$(pwd)/cxx.sh"
+        export LD="$(pwd)/ld.sh"
+
+        echo "#!/usr/bin/env bash" >> test.sh
+        chmod +x test.sh
+        declare -px >> test.sh
+        cp test.sh cc.sh
+        cp test.sh cxx.sh
+        cp test.sh ld.sh
+
+        echo "(clang \"\$@\" ) || (clang++ \"\$@\" )" >> cc.sh
```

Comment:
Have you tried just doing "clang++" for this? I think that clang++ should support both (although some of the Nix purity stuff may break this).

Reply:
I initially tried this, but it caused failures when building a few dependencies. I'll try to find a log so I can at least put a comment explaining it.